### ISSUE TITLE
Project level 2

### ISSUE THREAD

**Issue by fooheeyuan** [nus-cs2030/2021-s2#139]
Do we have to pass all the stages at the same time? My issue is that for each level, the inputs change which means we need to change the classes which will affect the next level. I tried to fix it by creating separate folders of all the classes to run for each MainN, but Code crunch doesn't accept folders. Maybe I should create multiple copies of the same class for each level eg Customer1 for Main1 etc..? Or is there a better way to go about doing this?

**Comment by eltonngkk** [comment]
Hi, I believe you should overload your constructors for each class to accept different parameters so that a single folder can run all your different MainN classes. It doesn't make sense to create multiple copies of each class. Goodluck!

**Comment by fooheeyuan** [comment]
> Hi, I believe you should overload your constructors for each class to accept different parameters so that a single folder can run all your different MainN classes. It doesn't make sense to create multiple copies of each class. Goodluck!

Hi thanks for the suggestion; the new constructors make sense to me but what if my attributes have been changed? 
1. Does this mean i just keep adding new attributes instead of replacing them
2. More importantly, my methods will have to change to change each attribute. Does this mean i also overload the methods? It basically means im just creating multiple copies of the same function which to me just makes it very complicated?? I think having different classes for different levels is more elegant and less likely to confuse everyone

**Comment by gableejh** [comment]
> Do we have to pass all the stages at the same time? My issue is that for each level, the inputs change which means we need to change the classes which will affect the next level. I tried to fix it by creating separate folders of all the classes to run for each MainN, but Code crunch doesn't accept folders. Maybe I should create multiple copies of the same class for each level eg Customer1 for Main1 etc..? Or is there a better way to go about doing this?

Yup you'll have to pass all stages, so for e.g. it evaluates level by level so if it doesn't pass level 1, level 3 won't be evaluated. You might have to overload the constructors. In the case of Level 1 and Level 2, you can update your Main1.java file to accept a service time as well, i.e. 1

For example,
Level 1: `new Customer(numOfCustomers, arrivalTime, 1)` 
Level 2: `new Customer(numOfCustomers, arrivalTime, serviceTime)`

**Comment by seancxy** [comment]
> > Hi, I believe you should overload your constructors for each class to accept different parameters so that a single folder can run all your different MainN classes. It doesn't make sense to create multiple copies of each class. Goodluck!
> 
> Hi thanks for the suggestion; the new constructors make sense to me but what if my attributes have been changed?
> 
> 1. Does this mean i just keep adding new attributes instead of replacing them
> 2. More importantly, my methods will have to change to change each attribute. Does this mean i also overload the methods? It basically means im just creating multiple copies of the same function which to me just makes it very complicated?? I think having different classes for different levels is more elegant and less likely to confuse everyone

you might not need to implement multiple copies of the same method if you define **default** values to take in if there are no values (service time is not in lv 1 etc.)

**Comment by eltonngkk** [comment]
> > Hi, I believe you should overload your constructors for each class to accept different parameters so that a single folder can run all your different MainN classes. It doesn't make sense to create multiple copies of each class. Goodluck!
> 
> Hi thanks for the suggestion; the new constructors make sense to me but what if my attributes have been changed?
> 
> 1. Does this mean i just keep adding new attributes instead of replacing them
> 2. More importantly, my methods will have to change to change each attribute. Does this mean i also overload the methods? It basically means im just creating multiple copies of the same function which to me just makes it very complicated?? I think having different classes for different levels is more elegant and less likely to confuse everyone

Yes, you need to keep adding attributes and assign them default values if your constructor does not take in that parameter. This will solve your 2nd issue as well.

**Comment by sebin0817** [comment]
> > Hi, I believe you should overload your constructors for each class to accept different parameters so that a single folder can run all your different MainN classes. It doesn't make sense to create multiple copies of each class. Goodluck!
> 
> Hi thanks for the suggestion; the new constructors make sense to me but what if my attributes have been changed?
> 
> 1. Does this mean i just keep adding new attributes instead of replacing them
> 2. More importantly, my methods will have to change to change each attribute. Does this mean i also overload the methods? It basically means im just creating multiple copies of the same function which to me just makes it very complicated?? I think having different classes for different levels is more elegant and less likely to confuse everyone

Not sure if you solved it yes, but i would advise against changing your attributes, as there should be minimal changes (best not at all) to your Main class(es) to pass each level. Like everyone said, you should overload constructors / methods so that different Main files can access the appropriate ones. all the best!
